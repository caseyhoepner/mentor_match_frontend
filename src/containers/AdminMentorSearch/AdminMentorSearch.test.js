import React from 'react';
import { shallow } from 'enzyme';
import { AdminMentorSearch, mapStateToProps, mapDispatchToProps } from './';

describe('AdminMentorSearch', () => {
  let wrapper;
  let mockFunc = jest.fn()
  let mockFunc2 = jest.fn()
  let mockFunc3 = jest.fn()
  
  beforeEach(() => {
    wrapper = shallow(<AdminMentorSearch
      setSearch={mockFunc}
      setLocale={mockFunc2}
      toggleShowingMentors={mockFunc3}
      showingAllMentors={false}
    />)
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange function', () => {
    it('should update search value state with the value passed in', () => {
      const mockEvent = {
        target: {
          name: 'search',
          value: 'skwijb'
        }
      }

      expect(wrapper.state().search).toEqual('')
      wrapper.instance().handleChange(mockEvent)
      expect(wrapper.state().search).toEqual('skwijb')
    });

    it('should fire setSearch with the proper params', () => {
      const mockEvent = {
        target: {
          name: 'search',
          value: 'skwijb'
        }
      }

      wrapper.instance().handleChange(mockEvent)
      expect(mockFunc).toHaveBeenCalledWith('skwijb')
    });

    it('should update localeSelected value state with the value passed in', () => {
      const mockEvent = {
        target: {
          name: 'localeSelected',
          value: 'Denver'
        }
      }

      expect(wrapper.state().localeSelected).toEqual('')
      wrapper.instance().handleChange(mockEvent)
      expect(wrapper.state().localeSelected).toEqual('Denver')
    });

    it('should fire setLocale with the proper params', () => {
      const mockEvent = {
        target: {
          name: 'localeSelected',
          value: 'Denver'
        }
      }

      wrapper.instance().handleChange(mockEvent)
      expect(mockFunc2).toHaveBeenCalledWith('Denver')
    });
  });

  describe('toggleClicked function', () => {
    it('should toggle preferencesClicked in state', () => {
      let mockEvent = {
        target: {
          name: 'preferencesClicked'
        }
      }

      expect(wrapper.state().preferencesClicked).toEqual(false)
      wrapper.instance().toggleClicked(mockEvent)
      expect(wrapper.state().preferencesClicked).toEqual(true)
    });

    it('should fire toggleShowingAllMentors with proper params', () => {
      let mockEvent = {
        target: {
          name: 'allMentorsClicked'
        }
      }

      wrapper.instance().toggleClicked(mockEvent)
      expect(mockFunc3).toHaveBeenCalledWith(true)
    });
  });
});

describe('mapStateToProps function', () => {
  it('should return an object with locale and showingAllMentors boolean', () => {
    const mockState = {
      locale: 'Denver',
      showingAllMentors: true,
      somethingElse: 'w00t!'
    }
    const expected = {
      locale: 'Denver',
      showingAllMentors: true
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps function', () => {
  const mockDispatch = jest.fn();
  let mappedProps;

  beforeEach(() => {
    mappedProps = mapDispatchToProps(mockDispatch)
  });

  it('should call dispatch when setLocale is called', () => {
    mappedProps.setLocale()
    expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call dispatch when setSeach is called', () => {
    mappedProps.setSearch()
    expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call dispatch when toggleShowingMentors is called', () => {
    mappedProps.toggleShowingMentors()
    expect(mockDispatch).toHaveBeenCalled()
  });
});