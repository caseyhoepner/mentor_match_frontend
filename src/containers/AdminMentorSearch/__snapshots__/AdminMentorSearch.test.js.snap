// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMentorSearch matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminMentorSearch
    setLocale={[MockFunction]}
    setSearch={[MockFunction]}
    showingAllMentors={false}
    toggleShowingMentors={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ams-search-and-buttons"
        >
          <div
            className="ams-search-inputs-container"
          >
            <div
              className="ams-search-bar"
            >
              <input
                className="ams-search-input"
                name="search"
                onChange={[Function]}
                placeholder="Search here..."
                type="text"
                value=""
              />
              <div
                className="ams-search-icon"
              />
            </div>
            <select
              className="ams-location-dropdown"
              name="localeSelected"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                --Select locale--
              </option>
              <option
                value="Denver"
              >
                Denver
              </option>
              <option
                value="Remote"
              >
                Remote
              </option>
            </select>
          </div>
          <div
            className="ams-filter-btn-container"
          >
            <button
              className="ams-filter-btn"
              name="preferencesClicked"
              onClick={[Function]}
            >
              Filter by Preferences
            </button>
            <button
              className="ams-filter-btn"
              name="allMentorsClicked"
              onClick={[Function]}
            >
              All Mentors
            </button>
          </div>
        </div>,
        <Connect(AdminPreferences)
          preferencesClicked={false}
        />,
      ],
      "className": "ams-search-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ams-search-inputs-container"
            >
              <div
                className="ams-search-bar"
              >
                <input
                  className="ams-search-input"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here..."
                  type="text"
                  value=""
                />
                <div
                  className="ams-search-icon"
                />
              </div>
              <select
                className="ams-location-dropdown"
                name="localeSelected"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  --Select locale--
                </option>
                <option
                  value="Denver"
                >
                  Denver
                </option>
                <option
                  value="Remote"
                >
                  Remote
                </option>
              </select>
            </div>,
            <div
              className="ams-filter-btn-container"
            >
              <button
                className="ams-filter-btn"
                name="preferencesClicked"
                onClick={[Function]}
              >
                Filter by Preferences
              </button>
              <button
                className="ams-filter-btn"
                name="allMentorsClicked"
                onClick={[Function]}
              >
                All Mentors
              </button>
            </div>,
          ],
          "className": "ams-search-and-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="ams-search-bar"
                >
                  <input
                    className="ams-search-input"
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here..."
                    type="text"
                    value=""
                  />
                  <div
                    className="ams-search-icon"
                  />
                </div>,
                <select
                  className="ams-location-dropdown"
                  name="localeSelected"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    --Select locale--
                  </option>
                  <option
                    value="Denver"
                  >
                    Denver
                  </option>
                  <option
                    value="Remote"
                  >
                    Remote
                  </option>
                </select>,
              ],
              "className": "ams-search-inputs-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      className="ams-search-input"
                      name="search"
                      onChange={[Function]}
                      placeholder="Search here..."
                      type="text"
                      value=""
                    />,
                    <div
                      className="ams-search-icon"
                    />,
                  ],
                  "className": "ams-search-bar",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "ams-search-input",
                      "name": "search",
                      "onChange": [Function],
                      "placeholder": "Search here...",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "ams-search-icon",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value=""
                    >
                      --Select locale--
                    </option>,
                    <option
                      value="Denver"
                    >
                      Denver
                    </option>,
                    <option
                      value="Remote"
                    >
                      Remote
                    </option>,
                  ],
                  "className": "ams-location-dropdown",
                  "name": "localeSelected",
                  "onChange": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "--Select locale--",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": "--Select locale--",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Denver",
                      "value": "Denver",
                    },
                    "ref": null,
                    "rendered": "Denver",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Remote",
                      "value": "Remote",
                    },
                    "ref": null,
                    "rendered": "Remote",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="ams-filter-btn"
                  name="preferencesClicked"
                  onClick={[Function]}
                >
                  Filter by Preferences
                </button>,
                <button
                  className="ams-filter-btn"
                  name="allMentorsClicked"
                  onClick={[Function]}
                >
                  All Mentors
                </button>,
              ],
              "className": "ams-filter-btn-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Filter by Preferences",
                  "className": "ams-filter-btn",
                  "name": "preferencesClicked",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Filter by Preferences",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "All Mentors",
                  "className": "ams-filter-btn",
                  "name": "allMentorsClicked",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "All Mentors",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "preferencesClicked": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ams-search-and-buttons"
          >
            <div
              className="ams-search-inputs-container"
            >
              <div
                className="ams-search-bar"
              >
                <input
                  className="ams-search-input"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here..."
                  type="text"
                  value=""
                />
                <div
                  className="ams-search-icon"
                />
              </div>
              <select
                className="ams-location-dropdown"
                name="localeSelected"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  --Select locale--
                </option>
                <option
                  value="Denver"
                >
                  Denver
                </option>
                <option
                  value="Remote"
                >
                  Remote
                </option>
              </select>
            </div>
            <div
              className="ams-filter-btn-container"
            >
              <button
                className="ams-filter-btn"
                name="preferencesClicked"
                onClick={[Function]}
              >
                Filter by Preferences
              </button>
              <button
                className="ams-filter-btn"
                name="allMentorsClicked"
                onClick={[Function]}
              >
                All Mentors
              </button>
            </div>
          </div>,
          <Connect(AdminPreferences)
            preferencesClicked={false}
          />,
        ],
        "className": "ams-search-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ams-search-inputs-container"
              >
                <div
                  className="ams-search-bar"
                >
                  <input
                    className="ams-search-input"
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here..."
                    type="text"
                    value=""
                  />
                  <div
                    className="ams-search-icon"
                  />
                </div>
                <select
                  className="ams-location-dropdown"
                  name="localeSelected"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    --Select locale--
                  </option>
                  <option
                    value="Denver"
                  >
                    Denver
                  </option>
                  <option
                    value="Remote"
                  >
                    Remote
                  </option>
                </select>
              </div>,
              <div
                className="ams-filter-btn-container"
              >
                <button
                  className="ams-filter-btn"
                  name="preferencesClicked"
                  onClick={[Function]}
                >
                  Filter by Preferences
                </button>
                <button
                  className="ams-filter-btn"
                  name="allMentorsClicked"
                  onClick={[Function]}
                >
                  All Mentors
                </button>
              </div>,
            ],
            "className": "ams-search-and-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="ams-search-bar"
                  >
                    <input
                      className="ams-search-input"
                      name="search"
                      onChange={[Function]}
                      placeholder="Search here..."
                      type="text"
                      value=""
                    />
                    <div
                      className="ams-search-icon"
                    />
                  </div>,
                  <select
                    className="ams-location-dropdown"
                    name="localeSelected"
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value=""
                    >
                      --Select locale--
                    </option>
                    <option
                      value="Denver"
                    >
                      Denver
                    </option>
                    <option
                      value="Remote"
                    >
                      Remote
                    </option>
                  </select>,
                ],
                "className": "ams-search-inputs-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="ams-search-input"
                        name="search"
                        onChange={[Function]}
                        placeholder="Search here..."
                        type="text"
                        value=""
                      />,
                      <div
                        className="ams-search-icon"
                      />,
                    ],
                    "className": "ams-search-bar",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "ams-search-input",
                        "name": "search",
                        "onChange": [Function],
                        "placeholder": "Search here...",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "ams-search-icon",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value=""
                      >
                        --Select locale--
                      </option>,
                      <option
                        value="Denver"
                      >
                        Denver
                      </option>,
                      <option
                        value="Remote"
                      >
                        Remote
                      </option>,
                    ],
                    "className": "ams-location-dropdown",
                    "name": "localeSelected",
                    "onChange": [Function],
                    "value": "",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "--Select locale--",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": "--Select locale--",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Denver",
                        "value": "Denver",
                      },
                      "ref": null,
                      "rendered": "Denver",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Remote",
                        "value": "Remote",
                      },
                      "ref": null,
                      "rendered": "Remote",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="ams-filter-btn"
                    name="preferencesClicked"
                    onClick={[Function]}
                  >
                    Filter by Preferences
                  </button>,
                  <button
                    className="ams-filter-btn"
                    name="allMentorsClicked"
                    onClick={[Function]}
                  >
                    All Mentors
                  </button>,
                ],
                "className": "ams-filter-btn-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Filter by Preferences",
                    "className": "ams-filter-btn",
                    "name": "preferencesClicked",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Filter by Preferences",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "All Mentors",
                    "className": "ams-filter-btn",
                    "name": "allMentorsClicked",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "All Mentors",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "preferencesClicked": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
